examples/pytorch/graphsage/experimental/train_dist_unsupervised.py

```diff
@@ -431,7 +434,7 @@ def main(args):
     global_valid_nid = th.LongTensor(np.nonzero(g.ndata['val_mask'][np.arange(g.number_of_nodes())]))
     global_test_nid = th.LongTensor(np.nonzero(g.ndata['test_mask'][np.arange(g.number_of_nodes())]))
     labels = g.ndata['labels'][np.arange(g.number_of_nodes())]
-    device = th.device('cpu')
+    device = th.device('cuda:' + str(g.rank() % 8))
```

Comment:
We cannot assume the number of gpus is 8.

Reply:
I think we should give user a parameter to set all the gpus like _--gpus 0,1,2,3,4,5,6,7_